modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerPipeliningTests.java

```diff
@@ -237,17 +245,15 @@ public void run() {
             final DefaultFullHttpResponse httpResponse = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK, buffer);
             httpResponse.headers().add(HttpHeaderNames.CONTENT_LENGTH, buffer.readableBytes());
 
-            final QueryStringDecoder decoder = new QueryStringDecoder(uri);
-
-            final int timeout =
-                uri.startsWith("/slow") && decoder.parameters().containsKey("sleep") ?
-                    Integer.valueOf(decoder.parameters().get("sleep").get(0)) : 0;
-            if (timeout > 0) {
+            final boolean slow = uri.matches("/slow/\\d+");
+            if (slow) {
                 try {
-                    Thread.sleep(timeout);
+                    Thread.sleep(scaledRandomIntBetween(500, 1000));
```

Comment:
Consider the case of two requests sent over the same connection. With HTTP pipelining enabled, we expect that the responses will come back in the order that they were sent. With HTTP pipelining disabled, we can make no such expectation, the results can come back in either possible order. That is, unless we artificially delay one of the responses. Thus, we have two endpoints, `/{id}` and `/slow/{id}`.

Then, with requests sent to `/slow/1` and `/2`, with pipelining enabled, we expect the results come back in the same order, even though `/slow/1` is going to be slow to process. With HTTP pipelining disabled, we expect the results to come back in the order `/2`, `/slow/1`. This shows that processing the slow endpoint is not blocking responses to the fast endpoint. 

With a 5--10 ms sleep, the sleep is not long enough to confidently make this assertion, the responses to the slow and fast endpoints can race.

> I don't think we should continue to go the path of long sleep-based tests

This is not a "long" sleep. With *no* sleep (and accepting the failing test runs), the max runtime on my machine is 2.31s. With the sleep that I have, the max runtime on my machine is 3.04s. Again, this is because the total sleep here is ~ the max of the sleeps across all the threads, and that is bounded by 1s and usually closer to 500ms because of the use of scaled random.